### ISSUE TITLE
Lower precision for HuggingFace Transformers

### ISSUE THREAD

**Issue by janandreschweiger** [UKPLab/sentence-transformers#1134]
Hey everyone,

I noticed that the floating point precision is considerably lower when using HuggingFace Transformers in contrast to using the SentenceTransformers library.

Due to our special setup we have to use HuggingFace for encoding the query and SentenceTransformers for encoding the documents. Do you think that this precision indifference will be a problem?

**Sample Output:**
```
Hugging Face:
tensor([ 0.0520,  0.0716, -0.6904, -0.2200, -0.1262,  0.6993,  0.8501,  0.1759, ...
Sentence Transformers:
[ 0.05200781  0.07160516 -0.6903786  -0.21998055 -0.12615614  0.6993378, ...


```

**Sample Code:**
```
from transformers import AutoTokenizer, AutoModel
import torch
from sentence_transformers import SentenceTransformer

if __name__ == '__main__':
    def mean_pooling(model_output, attention_mask):
        token_embeddings = model_output[0] #First element of model_output contains all token embeddings
        input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
        return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
    sentences = ['This is an example sentence', 'Each sentence is converted']

    tokenizer = AutoTokenizer.from_pretrained('sentence-transformers/msmarco-distilbert-multilingual-en-de-v2-tmp-lng-aligned')
    model = AutoModel.from_pretrained('sentence-transformers/msmarco-distilbert-multilingual-en-de-v2-tmp-lng-aligned')

    encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt')

    with torch.no_grad():
        model_output = model(**encoded_input)

    sentence_embeddings = mean_pooling(model_output, encoded_input['attention_mask'])

    print("Hugging Face:")
    print(sentence_embeddings[0])


    model = SentenceTransformer('sentence-transformers/msmarco-distilbert-multilingual-en-de-v2-tmp-lng-aligned')
    embeddings = model.encode(sentences)
    print("Sentence Transformers:")
    print(embeddings[0])
```

**Comment by nreimers** [comment]
This is just the output. The first is a pytorch tensor, which truncates the decimal points. Sentence transformers returns by default numpy, which prints more decimal points.

But the numbers are equal, just the output print is different for numpy and pytorch tensor

**Comment by janandreschweiger** [comment]
You are right @nreimers. Thank you.

Nevertheless, I opened this issue, because I compared the outputs after converting tensors to numpy arrays. As you can see they differ slightly. But I don't think that is that big of a deal:
```
Hugging Face:
[ 0.05200792  0.0716051  -0.69037884 -0.21998061 -0.12615621  0.6993378
  0.85009116  0.17590114 -0.60863745  0.19308855 -0.5854848  -0.5348679
Sentence Transformers:
[ 0.05200781  0.07160516 -0.6903786  -0.21998055 -0.12615614  0.6993378
  0.85009116  0.17590138 -0.6086378   0.19308843 -0.5854848  -0.5348678
```

**Comment by nreimers** [comment]
This is due to different padding behavior. It has no effect on downstream tasks.
